src/tiledb/cloud/utilities/consolidate.py

```diff
@@ -0,0 +1,248 @@
+from collections import defaultdict
+from os.path import basename
+from typing import Any, Mapping, Optional, Sequence
+
+import tiledb
+from tiledb.cloud import dag
+from tiledb.cloud.rest_api.models import RetryStrategy
+from tiledb.cloud.utilities import get_logger
+from tiledb.cloud.utilities import max_memory_usage
+from tiledb.cloud.utilities import run_dag
+
+MAX_FRAGMENT_SIZE = 1 << 30  # bytes
+
+
+def group_fragments(
+    array_uri: str,
+    *,
+    config: Optional[Mapping[str, Any]] = None,
+    group_first_dim: bool = True,
+) -> Sequence[Sequence[tiledb.FragmentInfo]]:
+    """
+    Get a list of fragment info objects, optionally grouping fragments that have the
+    same value for the first dimension.
+
+    :param array_uri: array URI
+    :param config: config dictionary, defaults to None
+    :param group_first_dim: group by first dimension, defaults to True
+    :return: list of lists of fragment info objects
+    """
+
+    logger = get_logger()
+    logger.info("Grouping fragments for array %r", array_uri)
+
+    with tiledb.scope_ctx(config):
+        fragment_info_lists = defaultdict(list)
+
+        fis = tiledb.FragmentInfoList(array_uri)
+
+        for fi in fis:
+            if group_first_dim:
+                fragment_info_lists[fi.nonempty_domain[0]].append(fi)
+            else:
+                fragment_info_lists["all"].append(fi)
+
+        # Create a list of lists of fragment info objects.
+        results = [x for x in list(fragment_info_lists.values())]
```

Comment:
This seems like it reifies the same thing into a list two times:

 1. `list(fragment_info_lists.values())`
 2. the no-op comprehension

It can be reduced to `results = list(fragment_info_lists.values())`.

(Did this maybe used to have some condition that got dropped?)

Reply:
Yes, this line went through a few changes. I'll simplify.